octopusdeploy/data_source_worker_pools.go

```diff
@@ -39,7 +40,10 @@ func dataSourceWorkerPoolsRead(ctx context.Context, d *schema.ResourceData, m in
 			return diag.FromErr(err)
 		}
 
-		flattenedWorkerPools = append(flattenedWorkerPools, flattenWorkerPool(workerPoolResource))
+		// Octopus currently doesn't support filtering by name, so we have to do it ourselves
+		if name != "" && workerPoolResource.Name == name {
+			flattenedWorkerPools = append(flattenedWorkerPools, flattenWorkerPool(workerPoolResource))
+		}
```

Comment:
I think this will break the behaviour for when people want to use `partial_name` as the filter - with no `name` value, the datasource will always return empty results.

What if we did this instead?

```suggestion
		// There is no name filter on the WorkerPools endpoint in the Octopus API, so filter in-memory if the name field is specified
		if name != "" && workerPoolResource.Name != name { continue } 
		
		flattenedWorkerPools = append(flattenedWorkerPools, flattenWorkerPool(workerPoolResource))
```

Reply:
You're absolutely right! Fixed this 😄

Reply:
There are some quirks of the current shape of our build/test pipeline that mean we currently need to merge to a branch within the main repo first, so we'll do that and update here when it gets released.